ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/submission/PartySetsITSpec.scala

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.benchtool.submission
+
+import com.daml.ledger.api.benchtool.{BenchtoolSandboxFixture, ConfigEnricher}
+import com.daml.ledger.api.benchtool.config.WorkflowConfig
+import com.daml.ledger.api.benchtool.config.WorkflowConfig.FooSubmissionConfig.{
+  ConsumingExercises,
+  NonconsumingExercises,
+  PartySet,
+}
+import com.daml.ledger.api.benchtool.config.WorkflowConfig.StreamConfig.{
+  PartyNamePrefixFilter,
+  TransactionTreesStreamConfig,
+}
+import com.daml.ledger.api.benchtool.services.LedgerApiServices
+import com.daml.ledger.api.testing.utils.SuiteResourceManagementAroundAll
+import com.daml.ledger.api.v1.ledger_offset.LedgerOffset
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.{AppendedClues, OptionValues}
+
+import scala.concurrent.Future
+
+class PartySetsITSpec
+    extends AsyncFlatSpec
+    with BenchtoolSandboxFixture
+    with SuiteResourceManagementAroundAll
+    with Matchers
+    with AppendedClues
+    with OptionValues {
+
+  it should "submit a party-set and apply party-set filter on a stream" in {
+    val submissionConfig = WorkflowConfig.FooSubmissionConfig(
+      numberOfInstances = 10,
+      numberOfObservers = 1,
```

Comment:
I'm not sure about how this test is represented, reading numberOfObservers = 1 and then observerPartySet0.count = 100 seems too confusing.

Could we get away with a single observer definition ?

Reply:
I did it this way to keep backwards compatibility (avoiding needing to update all existing workflow configs)
and to support a slightly different semantic for the party set:
- subsequent "regular" observers have exponentially smaller (hardcoded) chances of being picked for a contract observer,
- all parties from a party set have the same chances of being picked and the exact probability is configurable.

The way to think about this is a contract can get both some "regular" observers and some party set observers.

Reply:
> avoiding needing to update all existing workflow configs
The problem with having so many configs is that they make changing the config (for the better) much harder.  Maybe a script that generates the multiple variations is the way forward (so that we can put change the generation in one place).

> The way to think about this is a contract can get both some "regular" observers and some party set observers.
This is quite subtle (for the next reader).  Maybe the way to resolve this is to use names that reflect this (`exponentialProbabiltyObservers`, `fixedProbabilityObservers`) ?